crowbar_framework/app/models/service_object.rb

```diff
@@ -722,14 +725,14 @@ def proposal_create(params)
     # When we create a proposal, it might be "invalid", as some roles might be missing
     # This is OK, as the next step for the user is to add nodes to the roles
     # But we need to skip the after_save validations in the _proposal_update
-    _proposal_update(proposal, false)
+    _proposal_update(@bc_name, params["id"] || params[:name], proposal, false)
```

Comment:
Hrm, it seems the beginning of the method assumes `params["id"]` is always non-nil. Not that it matters much (and I think your bit is actually more correct based on what everything else is doing)

Would it make sense to use the `base_id` variable instead? (and maybe change how it's defined)